app/api/dao/organization.py

```diff
@@ -0,0 +1,104 @@
+
+from http import HTTPStatus
+from flask import json
+from app.database.models.bit_schema.organization import OrganizationModel
+from app.database.models.bit_schema.user_extension import UserExtensionModel
+from app import messages
+from app.api.request_api_utils import AUTH_COOKIE
+from app.utils.bitschema_utils import Timezone, OrganizationStatus
+
+
+class OrganizationDAO:
+    
+    """Data Access Object for Users_Extension functionalities"""
+
+    @staticmethod
+    def get_organization(rep_id):
+        """Retrieves the organization that is represented by the user which ID is passed as parameter.
+
+        Arguments:
+            rep_id: The ID of the user who represents the organization.
+
+        Returns:
+            The OrganizationModel class represented by the user whose ID was searched.
+        """
+
+        rep_additional_info = UserExtensionModel.find_by_user_id(rep_id)
+        try:
+            if rep_additional_info.is_organization_rep:
+                result = OrganizationModel.find_by_representative(rep_id)
+                if not result:
+                    return messages.ORGANIZATION_DOES_NOT_EXIST, HTTPStatus.NOT_FOUND
+                return {
+                    "rep_id": result.rep_id,
+                    "rep_department": result.rep_department,
+                    "organization_name":result.name, 
+                    "email": result.email,
+                    "about": result.about,
+                    "address": result.address,
+                    "website": result.website,
+                    "timezone": result.timezone.value,
+                    "phone": result.phone,
+                    "status": result.status.value,
+                    "join_date": result.join_date
+                }
+        except AttributeError as e:
+            logging.fatal(f"{e}")
+        return messages.IS_ORGANIZATION_REP_NOT_DECLARED, HTTPStatus.FORBIDDEN
```

Comment:
what does it mean? IS_ORGANIZATION_REP_NOT_DECLARED? can we name this better?

Reply:
https://www.codingblocks.net/podcast/clean-code-writing-meaningful-names/ - whenever you're free, Uncle Bob's clean code posts/podcasts are available. They should explain how important it is to have meaningful names

Reply:
> what does it mean? IS_ORGANIZATION_REP_NOT_DECLARED? can we name this better?

This means the user either has not declared whether or not they are an organization representative or has not yet created an additional_info and declare they are a representative. Tbh I can't think of any other name that is more suitable to be the title for this message. Do you have a suggestion @meenakshi-dhanani ?

Reply:
NO_REPRESENTATIVE_FOR_ORGANIZATION?

Reply:
Ok, will change it.